lib/active_merchant/billing/gateways/barclaycard_smartpay.rb

```diff
@@ -34,7 +34,7 @@ def authorize(money, creditcard, options = {})
         post[:card] = credit_card_hash(creditcard)
 
         if address = (options[:billing_address] || options[:address])
-          post[:billingAddress] = address_hash(address)
+          post[:card][:billingAddress] = address_hash(address)
```

Comment:
I had changed this when I began working on the update a month ago and honestly don't know why.  Remote tests pass without this nesting, so I will remove the change.